### PR TITLE
Remove unnecessary `ignores` for lint

### PR DESCRIPTION
**Pull request by fealho**
This PR removes lint rules from the `ignore` list in the `setup.cfg` file, as was done in our other libraries. The removed rules (D203, D213, D413, D406, W503) don't require any code change. The rule (SFS2), however, doesn't allow the usage of `format` strings, so those instances were updated to f-strings throughout the code.

It also removes `pylint`, since it doesn't seem to be used, and reorders the style check libraries in `setup.py`, removing repeated entries.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -53,17 +55,4 @@ collect_ignore = ['setup.py']
 
 [pydocstyle]
 convention = google
-add-ignore = D105, D107, D407, D203, D213, D413, D406, D417
-
-[pylint]
```
Comment (fealho) [inline setup.cfg:58]: @pvk-developer @amontanez24 Any idea why this is here? It doesn't seem to be used.

Comment (pvk-developer) [reply setup.cfg:58]: I don't recall having `pylint` on `SDV`.

**Review by pvk-developer — APPROVED** [review]

File: setup.cfg (hunk 1 above)
Comment (amontanez24) [reply setup.cfg:58]: it comes from this PR https://github.com/sdv-dev/SDV/commit/6ed01d67a5670b511284b21f94f0ccf712af1a0d

**Review by amontanez24 — APPROVED** [review]
LGTM :shipit:

**Pull request merged**